### PR TITLE
[BENCHMARK] Run parameter benchmarks with different input data and update them.

### PR DESCRIPTION
**Pull request by Irallia**
# min_var_length = 30 (remains)

old  value | new value
:---------:|:----------:
min_var_length results all | min_var_length results all
min_var_length results all | min_var_length results all
min_var_length results all | min_var_length results all
min_var_length results all | min_var_length results all
min_var_length results all | min_var_length results all

# max_var_length = 10.000 (changed from 100.000)

old  value | new value
:---------:|:----------:
max_var_length results all | max_var_length results all
max_var_length results all | max_var_length results all
max_var_length results all | max_var_length results all
max_var_length results all | max_var_length results all
max_var_length results all | max_var_length results all

# max_tol_inserted_length = 50 (remains)

old  value | new value
:---------:|:----------:
max_tol_inserted_length results all | max_tol_inserted_length results all
max_tol_inserted_length results all | max_tol_inserted_length results all
max_tol_inserted_length results all | max_tol_inserted_length results all
max_tol_inserted_length results all | max_tol_inserted_length results all
max_tol_inserted_length results all | max_tol_inserted_length results all

# max_tol_deleted_length = 50 (remains)

old  value | new value
:---------:|:----------:
max_tol_deleted_length results all | max_tol_deleted_length results all
max_tol_deleted_length results all | max_tol_deleted_length results all
max_tol_deleted_length results all | max_tol_deleted_length results all
max_tol_deleted_length results all | max_tol_deleted_length results all
max_tol_deleted_length results all | max_tol_deleted_length results all

# max_overlap = 50 (changed from 10)

old  value | new value
:---------:|:----------:
max_overlap results all | max_overlap results all
max_overlap results all | max_overlap results all
max_overlap results all | max_overlap results all
max_overlap results all | max_overlap results all
max_overlap results all | max_overlap results all

# partition_max_distance = 50 (changed from 1.000)

old  value | new value
:---------:|:----------:
partition_max_distance results all | partition_max_distance results all
partition_max_distance results all | partition_max_distance results all
partition_max_distance results all | partition_max_distance results all
partition_max_distance results all | partition_max_distance results all
partition_max_distance results all | partition_max_distance results all

# hierarchical_clustering_cutoff = 0.3 (changed from 0.5)

old  value | new value
:---------:|:----------:
hierarchical_clustering_cutoff results all | hierarchical_clustering_cutoff results all
hierarchical_clustering_cutoff results all | hierarchical_clustering_cutoff results all
hierarchical_clustering_cutoff results all | hierarchical_clustering_cutoff results all
hierarchical_clustering_cutoff results all | hierarchical_clustering_cutoff results all 
hierarchical_clustering_cutoff results all | hierarchical_clustering_cutoff results all

### CODE REVIEW

**Comment by Irallia** [comment]
# iGenVar only - new Plots:

old  values | new values
:----------:|:-----------:
iGenVar_only-results all | iGenVar_only-results all
iGenVar_only-results DUP_as_INS all | iGenVar_only-results DUP_as_INS all

File: test/benchmark/parameter_benchmarks/Snakefile [hunk 1]
```diff
@@ -83,35 +115,49 @@ rule reformat_truvari_results:
...
+        input6 = expand("results/parameter_benchmarks/{{dataset}}/truvari/partition_max_distance/{parameter_value}_min_qual_{min_qual}/pr_rec.txt",
+                parameter_value=[10, 25, 50, 100, 1000],        # default: 50
+                min_qual=list(range(config["quality_ranges"]["iGenVar"]["from"],
+                                    config["quality_ranges"]["iGenVar"]["to"],
+                                    config["quality_ranges"]["iGenVar"]["step"]))),
+        input7 = expand("results/parameter_benchmarks/{{dataset}}/truvari/hierarchical_clustering_cutoff/{parameter_value}_min_qual_{min_qual}/pr_rec.txt",
+                parameter_value= [0.1, 0.2, 0.3, 0.4],          # default: 0.3
+                min_qual=list(range(config["quality_ranges"]["iGenVar"]["from"],
+                                    config["quality_ranges"]["iGenVar"]["to"],
+                                    config["quality_ranges"]["iGenVar"]["step"])))
```
Comment (joergi-w) [inline test/benchmark/parameter_benchmarks/Snakefile:152]: This is copied for in each input – is there a way to pre-define this?

File: test/benchmark/parameter_benchmarks/Snakefile [hunk 2]
```diff
@@ -2,38 +2,69 @@ configfile: "Repos/iGenVar/test/benchmark/config/caller_comparison_config.yaml"
...
+                dataset=["Illumina_Paired_End", "Illumina_Mate_Pair", "MtSinai_PacBio", "PacBio_CCS", "10X_Genomics"],
+                parameter_name="max_tol_deleted_length"),
+        expand("results/parameter_benchmarks/{dataset}/plots/{parameter_name}.results.all.png",
+                dataset=["Illumina_Paired_End", "Illumina_Mate_Pair", "MtSinai_PacBio", "PacBio_CCS", "10X_Genomics"],
+                parameter_name="max_overlap"),
+        expand("results/parameter_benchmarks/{dataset}/plots/{parameter_name}.results.all.png",
+                dataset=["Illumina_Paired_End", "Illumina_Mate_Pair", "MtSinai_PacBio", "PacBio_CCS", "10X_Genomics"],
+                parameter_name="partition_max_distance"),
+        expand("results/parameter_benchmarks/{dataset}/plots/{parameter_name}.results.all.png",
+                dataset=["Illumina_Paired_End", "Illumina_Mate_Pair", "MtSinai_PacBio", "PacBio_CCS", "10X_Genomics"],
```
Comment (joergi-w) [inline test/benchmark/parameter_benchmarks/Snakefile:24]: the dataset is also always the same

**Review by joergi-w — APPROVED** [review]
Looks good! 😄 
Two ideas for the code:

File: test/benchmark/parameter_benchmarks/Snakefile (hunk 2 above)
Comment (Irallia) [reply test/benchmark/parameter_benchmarks/Snakefile:24]: Good point, this appears in every benchmark workflow. So I think I will try to make it shorter in another PR.

**Review by joshuak94 — APPROVED** [review]
Looks good! How did you come up with the new values?

**Comment by Irallia** [comment]
> Looks good! How did you come up with the new values?

With exactly this PR. So far I have tested iGenvar only on one dataset regarding the parameters, now I have added others. And then I changed only one parameter at a time and saw how our result behaves. With some back and forth I have now ended up with the dafault values and plots created here. On the plots you can see that larger or smaller values would not give any improvement.

**Pull request merged**